src/domains.cpp

```diff
@@ -104,10 +104,19 @@ unsigned int check_local_entropy (dolfin::Function *cation,
                                   double anion_valency,
                                   dolfin::Function *voltage,
                                   dolfin::Mesh *target_mesh,
-                                  double entropy_tol)
+                                  double entropy_tol,
+                                  int Max_Numb_Cells)
 {
+
   // compute mesh from input voltage function and transfer
   dolfin::Mesh mesh( *(voltage->function_space()->mesh()) );
+
+  if (mesh.num_cells()>Max_Numb_Cells){
```

Comment:
perhaps we can pass a negative number or some flag if we do not want to restrict the number of cells.